### ISSUE TITLE
How to fetchOrderbook using Proxy

### ISSUE THREAD

**Issue by jackpotcityco** [ccxt/ccxt#11834]
OS: Windows Server 2019 Datacenter
Programming Language version: Node.js 15.3.0
CCXT version: 1.49.98
Exchange: Binance and All Exchanges
Method: exchange.fetchOrderBook()

Hello!

I am trying to `fetchOrderBook` using the proxy method but are not successfull when doing the request. The proxy IP I use must use: **username, password, proxyIP and port** which I have just confirmed are a working proxyIP in another C# application. 
_(I have changed those for this post as seen)_

I am not sure if I miss something of how to set this up?

Thank you!

**This is the code I am trying to use:**
```     
        var HttpsProxyAgent = require('https-proxy-agent');
        
        var username = 'myusername';
        var password = 'mypassword';
        var proxyip = '209.197.291.180';
        var port = '9279';
        var proxy = 'https://' + username + ':' + password + '@' + proxyip + ':' + port; 
        var agent = new HttpsProxyAgent(proxy);

        const exchange = new ccxt.binance({ enableRateLimit: true, agent, 'timeout': 10000 });
        exchange.verbose = true;

        const orderbook = await exchange.fetchOrderBook("BTC/USDT", 5);
        console.log(orderbook);
```

-----------------------------------------------------------------------------------------------------------------------
**This is the error response I get:** (with `exchange.verbose = true`)

```
C:\myproject\node_modules\ccxt\js\base\Exchange.js:401
throw new ExchangeNotAvailable ([ this.id, method, url, e.type, e.message ].join (' '))
                                  ^

ExchangeNotAvailable: binance GET https://api.binance.com/api/v3/exchangeInfo system request to https://api.binance.com/api/v3/exchangeInfo failed, reason: write EPROTO 148:error:1408F10B:SSL routines:ssl3_get_record:wrong version number:c:\ws\deps\openssl\openssl\ssl\record\ssl3_record.c:332:

    at C:\myproject\node_modules\ccxt\js\base\Exchange.js:401:35
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (node:internal/process/task_queues:93:5)
    at async timeout (C:\myproject\node_modules\ccxt\js\base\functions\time.js:195:24)
    at async binance.request (C:\myproject\node_modules\ccxt\js\binance.js:3145:26)
    at async binance.fetchMarkets (C:\myproject\node_modules\ccxt\js\binance.js:828:26)
    at async binance.loadMarketsHelper (C:\myproject\node_modules\ccxt\js\base\Exchange.js:670:25)
    at async binance.fetchOrderBook (C:\myproject\node_modules\ccxt\js\binance.js:1273:9)
    at async testorderbooks (C:\myproject\functions.js:1116:27)
    at async testCommandStation (C:\myproject\functions.js:1079:5) {
  constructor: [class ExchangeNotAvailable extends NetworkError]
```

**Comment by kroitor** [comment]
It says you have an SSL version mistmatch:

- ExchangeNotAvailable: binance GET https://api.binance.com/api/v3/exchangeInfo system request to https://api.binance.com/api/v3/exchangeInfo failed, reason: write EPROTO 148:error:1408F10B:SSL routines:ssl3_get_record:wrong version number:c:\ws\deps\openssl\openssl\ssl\record\ssl3_record.c:332:

Have you tried testing the proxy with cURL in command line?
What is your version of openssl ?

Let us know if this does not help:

- https://www.google.com/search?q=ssl3_get_record%3Awrong+version+number

**Comment by jackpotcityco** [comment]
> It says you have an SSL version mistmatch:
> 
> * ExchangeNotAvailable: binance GET https://api.binance.com/api/v3/exchangeInfo system request to https://api.binance.com/api/v3/exchangeInfo failed, reason: write EPROTO 148:error:1408F10B:SSL routines:ssl3_get_record:wrong version number:c:\ws\deps\openssl\openssl\ssl\record\ssl3_record.c:332:
> 
> Have you tried testing the proxy with cURL in command line? What is your version of openssl ?
> 
> Let us know if this does not help:
> 
> * https://www.google.com/search?q=ssl3_get_record%3Awrong+version+number
Thank you!

1. I tried testing the proxy with cURL in the command line and that did work.

2. I noticed that I didn't have openssl installed. Now I have installed it and the version says: **OpenSSL 1.1.1l  24 Aug 2021**

I tried to run the code again. But it seems that I have the wrong version. What is the version that I should have and do you know of any link from where to install this?

**Comment by kroitor** [comment]
@jackpotcityco what is your version of `certifi` ? 
Please, paste `pip3 list` here.

**Comment by jackpotcityco** [comment]
> @jackpotcityco what is your version of `certifi` ? Please, paste `pip3 list` here.

I am trying google how to check this version. It seems to be python related?
I am not sure how to check this `certifi` version. Should I do that in the command prompt and how to get a `pip3 list`? 
_(I am sorry for not knowing how to check this)_

**Comment by kroitor** [comment]
@jackpotcityco pardon, i confused it with python, my bad. For `node`, please, upgrade your version of CCXT first (the most recent version is 1.72+). Let us know if that does not help

**Comment by jackpotcityco** [comment]
No problem, I updated the version to: `1.72.5`. I tried the code again and still receives the below error:
```
C:\myproject\node_modules\ccxt\js\base\Exchange.js:469
                            throw new ExchangeNotAvailable ([ this.id, method, url, e.type, e.message ].join (' '))
                                  ^

ExchangeNotAvailable: binance GET https://api.binance.com/api/v3/exchangeInfo system request to https://api.binance.com/api/v3/exchangeInfo failed, reason: write EPROTO 15184:error:1408F10B:SSL routines:ssl3_get_record:wrong version number:c:\ws\deps\openssl\openssl\ssl\record\ssl3_record.c:332:

    at C:\myproject\node_modules\ccxt\js\base\Exchange.js:469:35
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (node:internal/process/task_queues:93:5)
    at async timeout (C:\myproject\node_modules\ccxt\js\base\functions\time.js:203:20)
    at async binance.request (C:\myproject\node_modules\ccxt\js\binance.js:5175:26)
    at async binance.fetchMarkets (C:\myproject\node_modules\ccxt\js\binance.js:1366:26)
    at async binance.loadMarketsHelper (C:\myproject\node_modules\ccxt\js\base\Exchange.js:780:25)
    at async binance.fetchOrderBook (C:\myproject\node_modules\ccxt\js\binance.js:1914:9)
    at async testorderbooks (C:\myproject\functions.js:1117:27)
    at async testCommandStation (C:\myproject\functions.js:1080:5) {
  constructor: [class ExchangeNotAvailable extends NetworkError]
}
```

**Comment by jackpotcityco** [comment]
I think I did find the problem. It seems that the `https://` is wrong, it should be `http://`. The proxy itself should be accessed by HTTP and not HTTPS. So I changed it to this and now it was possible to return the orderbook.

`var proxy = 'http://' + username + ':' + password + '@' + proxyip + ':' + port;`

Thank you very much for your help and now I have the newest version of `ccxt` also.

Thank you!
